pkg/vmcp/client/client.go

```diff
@@ -239,6 +241,60 @@ func (h *httpBackendClient) defaultClientFactory(ctx context.Context, target *vm
 	return c, nil
 }
 
+// wrapBackendError wraps an error with the appropriate sentinel error based on error type.
+// This enables type-safe error checking with errors.Is() instead of string matching.
+//
+// Error detection strategy (in order of preference):
+// 1. Check for standard Go error types (context errors, net.Error, url.Error)
+// 2. Fall back to string pattern matching for library-specific errors (MCP SDK, HTTP libs)
+func wrapBackendError(err error, backendID string, operation string) error {
+	if err == nil {
+		return nil
+	}
+
+	// 1. Type-based detection: Check for context deadline/cancellation
+	if errors.Is(err, context.DeadlineExceeded) {
+		return fmt.Errorf("%w: failed to %s for backend %s (timeout): %v",
```

Comment:
Noticed that the original error gets appended with `%v` here, so it won't be in the error chain. Callers can check `errors.Is(err, vmcp.ErrTimeout)` but won't be able to dig into the underlying error type with `errors.As()`. Might be worth a quick doc comment mentioning this, or we could look at a custom error type if we need both later.